src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service;
+
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Map<String, Integer> fruits = new HashMap<>();

```

Comment:
You should not store the state of the fruits in the service class. This violates the Single Responsibility Principle as this class is now responsible for both manipulating the fruits and storing their state. It's better to store the state of the fruits in a separate class, for example, a `Storage` class. This class can then be used by other classes if they need to access the state of the fruits.